Flagsmith.FlagsmithClient/Trait.cs

```diff
@@ -6,63 +6,31 @@ namespace Flagsmith
     public class Trait
     {
         public Trait() { }
-        public Trait(string traitKey, string traitValue)
+        public Trait(string traitKey, dynamic traitValue)
         {
-            this.key = traitKey;
+            this.traitKey = traitKey;
             this.traitValue = traitValue;
         }
-        [JsonProperty("trait_key")]
-        private string key = null;
-
-        [JsonProperty("string_value")]
-        private string strValue = null;
-
-        [JsonProperty("integer_value")]
-        private int intValue = 0;
 
-        [JsonProperty("bool_value")]
-        private bool boolValue = false;
+        [JsonProperty("trait_key")]
+        private string traitKey = null;
 
-        // Support for legacy non-typed value (only support string)
         [JsonProperty("trait_value")]
-        private string traitValue = null;
+        private dynamic traitValue = null;
 
-        public string GetKey()
-        {
-            return key;
-        }
-
-        public string GetValue()
-        {
-            if (traitValue != null)
-            {
-                return traitValue;
-            }
-            return strValue;
-        }
-
-        public string GetStringValue()
-        {
-            if (traitValue != null)
-            {
-                return traitValue;
-            }
-            return strValue;
-        }
-
-        public bool GetBoolValue()
+        public override string ToString()
         {
-            return boolValue;
+            return JsonConvert.SerializeObject(this);
         }
 
-        public int GetIntValue()
+        public string getTraitKey()
         {
-            return intValue;
+            return this.traitKey;
         }
 
-        public override string ToString()
```

Comment:
Is this idiomatic c# to have public method use camel case instead of Pascal case?(not even sure if it's a method, though)

Reply:
Good question. I think you're right. I will use Pascal case.